### ISSUE TITLE
Error: Wrapped: Verification Failed for captain.renamed-domain.com

### ISSUE THREAD

**Issue by FloMaetschke** [caprover/caprover-cli#48]
After upgrading from verison 1.6.0 to 1.6.1 caprover did not recover.

I tried reinstalling with caprover 1.6.0 and 1.6.1 both do no longer work. It seems the certbot stuff is broken?

I found the other two issues about the Verification Failed problem, but both have no solution.

Interestingly the server running docker is hosted by hetzner, too.

Any help would be nice!

```
root@Ubuntu-1804-bionic-64-minimal ~ # docker logs 314

> caprover@0.0.0 start /usr/src/app
> node ./bin/www

Captain Starting ...
February 10th 2020, 8:40:11.129 am    Emptying generated and temp folders.
February 10th 2020, 8:40:11.138 am    Ensuring directories are available on host. Started.
February 10th 2020, 8:40:11.241 am    Ensuring directories are available on host. Finished.
February 10th 2020, 8:40:11.248 am    Network captain-overlay-network is already attached to service: captain-captain
February 10th 2020, 8:40:11.256 am    captain-captain (service) has already been connected to secret: captain-salt
February 10th 2020, 8:40:11.266 am    Copying fake certificates...
February 10th 2020, 8:40:11.271 am    Setting up NGINX conf file...
February 10th 2020, 8:40:11.274 am    Locking NGINX configuration reloading...
February 10th 2020, 8:40:11.296 am    SUCCESS: UNLocking NGINX configuration reloading...
February 10th 2020, 8:40:11.299 am    Captain Nginx is already running..
February 10th 2020, 8:40:11.302 am    Updating NGINX service...
February 10th 2020, 8:40:11.349 am    Waiting for 5 seconds for nginx reload to take into effect
February 10th 2020, 8:40:16.352 am    NGINX is fully set up and working...
February 10th 2020, 8:40:16.354 am    Migration not needed, skipping.
February 10th 2020, 8:40:16.359 am    Captain Certbot is already running..
February 10th 2020, 8:40:16.363 am    Updating Certbot service...
February 10th 2020, 8:40:16.461 am    executeCommand Container: captain-certbot certbot certificates --non-interactive
February 10th 2020, 8:40:18.629 am    **** Captain is initialized and ready to serve you! ****
February 10th 2020, 8:40:19.638 am    Error: Wrapped: Verification Failed for captain.renamed-domain.com
Error: Wrapped: Verification Failed for captain.renamed-domain.com
    at errorize (/usr/src/app/built/utils/Logger.js:9:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:31:19)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:61:32)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:392:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
February 10th 2020, 8:40:19.639 am    Error: Wrapped: Error        Error: connect ECONNREFUSED 123.123.123.123:80
Error: Wrapped: Error        Error: connect ECONNREFUSED 123.123.123.123:80
    at errorize (/usr/src/app/built/utils/Logger.js:9:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:31:19)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:62:32)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:392:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
February 10th 2020, 8:40:19.639 am    Error: Wrapped: body         undefined
Error: Wrapped: body         undefined
    at errorize (/usr/src/app/built/utils/Logger.js:9:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:31:19)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:63:32)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:392:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
February 10th 2020, 8:40:19.639 am    Error: Wrapped: randomUuid   3d1b2ad4-aeb1-46e4-b8d2-d6d029484b2e
Error: Wrapped: randomUuid   3d1b2ad4-aeb1-46e4-b8d2-d6d029484b2e
    at errorize (/usr/src/app/built/utils/Logger.js:9:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:31:19)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:64:32)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:392:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
February 10th 2020, 8:40:19.640 am    Captain health check failed: #0 at captain.renamed-domain.com
February 10th 2020, 8:40:19.640 am    NGINX health check failed: #0
February 10th 2020, 8:40:40.653 am    Error: Wrapped: Verification Failed for captain.renamed-domain.com
Error: Wrapped: Verification Failed for captain.renamed-domain.com
    at errorize (/usr/src/app/built/utils/Logger.js:9:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:31:19)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:61:32)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:392:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
February 10th 2020, 8:40:40.654 am    Error: Wrapped: Error        Error: connect ECONNREFUSED 123.123.123.123:80
Error: Wrapped: Error        Error: connect ECONNREFUSED 123.123.123.123:80
    at errorize (/usr/src/app/built/utils/Logger.js:9:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:31:19)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:62:32)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:392:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
February 10th 2020, 8:40:40.654 am    Error: Wrapped: body         undefined
Error: Wrapped: body         undefined
    at errorize (/usr/src/app/built/utils/Logger.js:9:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:31:19)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:63:32)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:392:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
February 10th 2020, 8:40:40.654 am    Error: Wrapped: randomUuid   cb905508-1e5d-4f1b-b598-f3c1bd1b9b00
Error: Wrapped: randomUuid   cb905508-1e5d-4f1b-b598-f3c1bd1b9b00
    at errorize (/usr/src/app/built/utils/Logger.js:9:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:31:19)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:64:32)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:392:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
February 10th 2020, 8:40:40.655 am    Captain health check failed: #1 at captain.renamed-domain.com
February 10th 2020, 8:40:40.655 am    NGINX health check failed: #1
February 10th 2020, 8:41:01.719 am    Error: Wrapped: Verification Failed for captain.renamed-domain.com
Error: Wrapped: Verification Failed for captain.renamed-domain.com
    at errorize (/usr/src/app/built/utils/Logger.js:9:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:31:19)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:61:32)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:392:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
February 10th 2020, 8:41:01.720 am    Error: Wrapped: Error        Error: connect ECONNREFUSED 123.123.123.123:80
Error: Wrapped: Error        Error: connect ECONNREFUSED 123.123.123.123:80
    at errorize (/usr/src/app/built/utils/Logger.js:9:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:31:19)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:62:32)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:392:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
February 10th 2020, 8:41:01.720 am    Error: Wrapped: body         undefined
Error: Wrapped: body         undefined
    at errorize (/usr/src/app/built/utils/Logger.js:9:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:31:19)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:63:32)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:392:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
February 10th 2020, 8:41:01.720 am    Error: Wrapped: randomUuid   b580ba99-6621-4ba1-a251-017a6c5e53e2
Error: Wrapped: randomUuid   b580ba99-6621-4ba1-a251-017a6c5e53e2
    at errorize (/usr/src/app/built/utils/Logger.js:9:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:31:19)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:64:32)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:392:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
February 10th 2020, 8:41:01.720 am    Captain health check failed: #2 at captain.renamed-domain.com
February 10th 2020, 8:41:01.721 am    NGINX health check failed: #2
February 10th 2020, 8:41:18.627 am    executeCommand Container: captain-certbot certbot certificates --non-interactive
February 10th 2020, 8:41:21.333 am    executeCommand Container: captain-certbot certbot renew --non-interactive
February 10th 2020, 8:41:22.767 am    Error: Wrapped: Verification Failed for captain.renamed-domain.com
Error: Wrapped: Verification Failed for captain.renamed-domain.com
    at errorize (/usr/src/app/built/utils/Logger.js:9:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:31:19)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:61:32)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:392:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
February 10th 2020, 8:41:22.768 am    Error: Wrapped: Error        Error: connect ECONNREFUSED 123.123.123.123:80
Error: Wrapped: Error        Error: connect ECONNREFUSED 123.123.123.123:80
    at errorize (/usr/src/app/built/utils/Logger.js:9:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:31:19)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:62:32)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:392:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
February 10th 2020, 8:41:22.768 am    Error: Wrapped: body         undefined
Error: Wrapped: body         undefined
    at errorize (/usr/src/app/built/utils/Logger.js:9:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:31:19)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:63:32)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:392:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
February 10th 2020, 8:41:22.768 am    Error: Wrapped: randomUuid   f1b6ebf5-b742-4aee-b806-76eb523737a9
Error: Wrapped: randomUuid   f1b6ebf5-b742-4aee-b806-76eb523737a9
    at errorize (/usr/src/app/built/utils/Logger.js:9:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:31:19)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:64:32)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:392:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
February 10th 2020, 8:41:22.769 am    Captain health check failed: #3 at captain.renamed-domain.com
February 10th 2020, 8:41:22.769 am    NGINX health check failed: #3
February 10th 2020, 8:41:44.800 am    Error: Wrapped: Verification Failed for captain.renamed-domain.com
Error: Wrapped: Verification Failed for captain.renamed-domain.com
    at errorize (/usr/src/app/built/utils/Logger.js:9:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:31:19)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:61:32)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:392:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
February 10th 2020, 8:41:44.801 am    Error: Wrapped: Error        Error: read ECONNRESET
Error: Wrapped: Error        Error: read ECONNRESET
    at errorize (/usr/src/app/built/utils/Logger.js:9:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:31:19)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:62:32)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:392:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
February 10th 2020, 8:41:44.801 am    Error: Wrapped: body         undefined
Error: Wrapped: body         undefined
    at errorize (/usr/src/app/built/utils/Logger.js:9:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:31:19)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:63:32)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:392:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
February 10th 2020, 8:41:44.801 am    Error: Wrapped: randomUuid   52481bfa-0d27-4519-b00a-51de88ed1054
Error: Wrapped: randomUuid   52481bfa-0d27-4519-b00a-51de88ed1054
    at errorize (/usr/src/app/built/utils/Logger.js:9:16)
    at Function.e (/usr/src/app/built/utils/Logger.js:31:19)
    at Request._callback (/usr/src/app/built/user/system/DomainResolveChecker.js:64:32)
    at self.callback (/usr/src/app/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:392:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
February 10th 2020, 8:41:44.802 am    Captain health check failed: #4 at captain.renamed-domain.com
February 10th 2020, 8:41:44.802 am    NGINX health check failed: #4
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! caprover@0.0.0 start: `node ./bin/www`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the caprover@0.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-02-10T08_41_44_817Z-debug.log

```

**Comment by FloMaetschke** [comment]
I cleaned the /captain folder after i did a backup from it.

left the docker swarm with:
docker swarm leave -f   (Forced) to remove all running docker instances

After that, i reinstalled caprover.
I had to manually add the apps, i was able to reuse the docker volumes.
Using /captain/data/config-captain.json as reference was a good help for the correct settings.

Everything works now again. 

Now i am looking for an automatic backup solution

Is saving /captain folder and /var/lib/docker/volumes enough?

I would love to see a solution like: 

copy all files and foldes in a temp backup dir and create a tar.gz with timestamp or something similiar.

recovery should work vice versa. Reinstall caprover, copy over everything and it must work again.
